tests/Value/Arithmetic/NumberTest.php

```diff
@@ -8,24 +8,24 @@
 
 final class NumberTest extends TestCase
 {
-    public function testAdd()
+    public function testAdd(): void
     {
         self::assertEquals(new Number(666), (new Number(123))->add(new Number(543)));
         self::assertTrue((new Number(666.0))->equals((new Number(123))->add(new Number(543))));
-        self::assertEquals(new Number(666.666), (new Number(123.123))->add(new Number(543.543)));
+        self::assertTrue((new Number(666.666))->equals((new Number(123.123, 14))->add(new Number(543.543))));
```

Comment:
Why add an explicit scale to this test-case? That should not be required for this case

Reply:
> All these alterations are made because the upgraded package seems to handle the rounding different from before (as you can see via the passing tests, even though this seems wrong).
> I haven't been able to pinpoint the exact cause, because the litipk/php-bignumbers package is >= 7.0, so the same version is used as before.

This will need to be looked at in detail then, because we do rely on this implicit natural scale behaviour. What is the result of the original test, without an explicit scale?

Reply:
> This will need to be looked at in detail then, because we do rely on this implicit natural scale behaviour. What is the result of the original test, without an explicit scale?

I think your earlier remark on the PHPUnit version is the missing piece in the puzzle.
There is an optional `$delta` parameter in the `assertEquals` method, which defaults to `0.0`, thereby implicitly ignoring the decimals. Looks like that behaviour is altered. (perhaps this issues has something to do with it: https://github.com/sebastianbergmann/phpunit/issues/3340).

> What is the result of the original test, without an explicit scale?

The result was that it expected `666.666` but received `666.66600000000000`. The explicit scale can be dropped here, because the issue is solved via `assertTrue` in combination with `equals`.